gradle/init.gradle.kts

```diff
@@ -1,32 +1,46 @@
-fun resolvePluginPath(rootPath: java.nio.file.Path): java.nio.file.Path? {
-    if (rootPath.resolve("src").toFile().exists()) {
-        return null
-    }
-    if (rootPath.resolve("demo").toFile().exists()) {
-        return null
-    }
-    var path = rootPath
-    for (n in 1..4) {
-        val settingsFile = path.resolve("demo")
-        if (settingsFile.toFile().exists()) {
-            return path.toAbsolutePath()
-            break
-        } else {
-            path = path.resolve("..")
-            println("new path: ${path.toAbsolutePath()}")
-        }
-    }
-    return null
-}
-
 gradle.settingsEvaluated {
-    val property = "org.jsonschema2dataclass.local"
-    if (extra.has(property) && extra[property].toString().toBoolean()) {
-        println("root dir: ${getRootDir()}")
-        val path = resolvePluginPath(getRootDir().toPath().toAbsolutePath())
-        if (path != null) {
-            println("Include build:\"$path\"")
-            includeBuild(path)
+    val localPublishExtra = "org.jsonschema2dataclass.internal.local-publish"
+    val localVersionExtra = "org.jsonschema2dataclass.internal.git-version"
+    if (extra.has(localPublishExtra) && extra[localPublishExtra].toString().toBoolean()) {
+        val localVersion = extra[localVersionExtra]!!.toString()
+        pluginManagement {
+            repositories {
+                mavenLocal().content {
```

Comment:
> How can I get a folder name into a github action variable to use it later? This is needed for publishing to local repository and I don't want to use a standard one.

I'm not sure what you mean here. There are many ways to set up a variable.
 * Hardcode it in the yaml file
 * [Use a GitHub Secret](https://user-images.githubusercontent.com/2906988/224508599-18f272bb-f284-487e-8ba6-a8f53dfa9c07.png)
 * You can change the .m2 folder location with settings.xml: https://maven.apache.org/settings.html#simple-values
 * You can probably change mavenLocal location in Gradle too
 * Register a new repository that's local:
```
publishing {
    repositories {
        maven("mavenInProject") {
            url = uri("${rootProject.projectDir}/build/repo")
        }
    }
}
```
and then `gradlew publish<Publication>ToMavenInProjectRepository` (`gradlew tasks` gives you a lot of help with this.

Reply:
I found, that results are not quite the same if you use `includeBuild` for the plugin or use a pre-built binary.

Thus I'd like to change `mavenLocal` repository location for demo using `maven.repo.local` system variable. This is the reason to write such complecated `init.gradle.kts` file.

I'll try various ideas and return with this later

Reply:
You might need to be explicit about substitutions:
https://docs.gradle.org/current/userguide/composite_builds.html#included_build_declaring_substitutions

It should be fully equivalent, to using the build version, unless the built version has some different flags on CI for example. Also have a look at `gradlew dependencies` to make sure the subs happened.

Reply:
Thank you. I've checked by adding `println("Hello world")` into a plugin :D 

I don't see any plugins listed in `gradlew dependencies`.

Reply:
The goal is to rewrite version used in project to version you specify by `-P<version.extra>` and without any modification of a project code. I use `setup-java` plugin and let it set up cache by checksum of all build and settings gradle files.

For plugins block you need to target pluginManagement and rewrite there, for buildscript.. buildscript need to be targeted ;P I've checked if there was a hit to original plugin by removing respective parts of gradle cache (`fd` is basically a ultra-fast `find` tool)

```sh
$ cd ~/.gradle
$ fd '.*jsonschema2dataclass.*' -X rm -rfv
``` 

This method works if it's written using plugins block:

```gradle
plugins {
  id("your.plugin.here") version "1.2.3"
}
```

or using a buildscript block:

```gradle
buildscript {
  repositories {
    maven {
      url = uri("https://plugins.gradle.org/m2/")
    }
  }
  dependencies {
    classpath("your.plugin.here:plugin:1.2.3")
  }
}

apply(plugin = "your.plugin.here")
```